tests/unit/test_util.py

```diff
@@ -97,6 +112,236 @@ def test_util_np_to_pil(
+                        [True, False, True, False],
+                    ],
+                    [
+                        [True, False, False, True],
+                        [True, True, True, False],
+                        [False, True, True, True],
+                        [True, False, True, True],
+                        [False, False, False, False],
+                    ],
+                    [
+                        [False, False, True, True],
+                        [False, True, False, False],
+                        [True, False, False, True],
+                        [True, False, False, False],
+                        [False, False, False, True],
+                    ],
+                ]
+            ),
+        ),
+        (
+            IMAGE2_RGBA,
+            37,
+            operator.lt,
+            np.array(
+                [
+                    [
+                        [False, False, False, False],
+                        [False, False, False, False],
+                        [False, False, False, False],
+                        [False, False, True, False],
+                        [False, False, False, False],
+                    ],
+                    [
+                        [True, False, True, False],
+                        [True, False, False, False],
+                        [False, False, False, True],
+                        [False, False, False, False],
+                        [False, False, True, False],
+                    ],
+                    [
+                        [False, False, False, False],
+                        [False, False, False, False],
+                        [True, False, False, False],
+                        [False, False, False, False],
+                        [False, False, True, True],
+                    ],
+                    [
+                        [False, False, False, False],
+                        [False, False, False, False],
+                        [True, True, True, False],
+                        [False, False, False, False],
+                        [False, False, False, False],
+                    ],
+                    [
+                        [True, False, False, False],
+                        [True, False, False, False],
+                        [False, False, False, False],
+                        [True, False, False, False],
+                        [False, False, False, False],
+                    ],
+                ]
+            ),
+        ),
+        (IMAGE3_RGBA_BLACK, 2, operator.gt, np.zeros((5, 5, 4), dtype=bool),),
+        (IMAGE3_RGBA_BLACK, 0, operator.lt, np.zeros((5, 5, 4), dtype=bool),),
+        (IMAGE3_RGBA_BLACK, 1, operator.lt, np.ones((5, 5, 4), dtype=bool),),
+        (IMAGE4_RGBA_WHITE, 0, operator.gt, np.ones((5, 5, 4), dtype=bool),),
+        (IMAGE4_RGBA_WHITE, 1, operator.lt, np.zeros((5, 5, 4), dtype=bool),),
+    ),
+)
+def test_util_threshold_to_mask(img, threshold, relate, expected_array):
```

Comment:
300 lines of fixtures before 3 lines of test if quite strange to read. In these cases I would prefere to have a separate fixture function like in test_slide